server/src/main/java/com/google/coffeehouse/servlets/JoinClubServlet.java

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.coffeehouse.storagehandler.StorageHandlerApi;
+import com.google.coffeehouse.common.MembershipConstants;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to join a club from Http POST Request Body (in JSON format) containing a club
+ * and user ID that exists in the database, and return a success status 200 (OK).
+ */
+@WebServlet("/api/join-club")
+public class JoinClubServlet extends HttpServlet {
+
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request cannot be used to join a club for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+  /** The message to be logged when the body of the POST request cannot be parsed. */
+  public static final String LOG_BODY_ERROR_MESSAGE = 
+      "Body unable to be parsed in JoinClubServlet: ";
+
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request does not have a required field.
+   */
+  public static final String NO_FIELD_ERROR = "No \"%s\" found in JSON.";
```

Comment:
nit: Consider updating error to say "Missing \"%s\" field in JSON" for clarity.

Reply:
Thanks for pointing this out!